api/src/services/lottery.service.ts

```diff
@@ -956,6 +974,64 @@ export class LotteryService {
     return formattedActivityLogs;
   }
 
+  /**
+    runs the job to auto expire lotteries that are passed their due date
+    will call the the cache purge to purge all listings as long as updates had to be made
+  */
+  async autoPublishResults(): Promise<SuccessDTO> {
+    this.logger.warn('autoPublishLotteryResults job running');
+    await this.listingService.markCronJobAsStarted(
+      LOTTERY_PUBLISH_CRON_JOB_NAME,
+    );
+    const tomorrow = dayjs(new Date()).add(1, 'days').toDate();
+    const releasedListings = await this.prisma.listings.findMany({
+      select: {
+        id: true,
+        name: true,
+        jurisdictions: true,
+      },
+      where: {
+        status: ListingsStatusEnum.closed,
+        reviewOrderType: ReviewOrderTypeEnum.lottery,
+        lotteryOptIn: true,
+        lotteryStatus: LotteryStatusEnum.releasedToPartners,
+        listingEvents: {
+          some: {
+            type: ListingEventsTypeEnum.publicLottery,
+            startDate: { lt: tomorrow },
+          },
+        },
+      },
+    });
+
+    await Promise.all(
+      releasedListings.map(async (listingRaw) => {
+        const listing = mapTo(Listing, listingRaw);
+        try {
+          await this.prisma.activityLog.create({
+            data: {
```

Comment:
Not really this ticket but what are your thoughts on having enums for module and action fields of the activity log? Especially as we start to use it on the FE it could be better to ensure consistency there. I could make a ticket but wanted to get your thoughts first.

Reply:
You KNOW I love enums. I support fully. Can bring it up at bloom eng workshop too if you want